scenarios/cpp/windows/captioning/captioning.cpp

```diff
@@ -0,0 +1,538 @@
+                {
+                    WriteToConsole("NOMATCH: Speech could not be recognized.\n");
+                }
+            });
+
+        speechRecognizer->Canceled.Connect([this, &recognitionEnd](const SpeechRecognitionCanceledEventArgs& e)
+            {
+                if (CancellationReason::EndOfStream == e.Reason)
+                {
+                    WriteToConsole("End of stream reached.\n");
+                    recognitionEnd.set_value(std::nullopt); // Notify to stop recognition.
+                }
+                else if (CancellationReason::CancelledByUser == e.Reason)
+                {
+                    WriteToConsole("User canceled request.\n");
+                    recognitionEnd.set_value(std::nullopt); // Notify to stop recognition.
+                }
+                else if (CancellationReason::Error == e.Reason)
+                {
+                    std::ostringstream error;
+                    error << "Encountered error.\n"
+                        << "ErrorCode: " << (int)e.ErrorCode << "\n"
+                        << "ErrorDetails: " << e.ErrorDetails << std::endl;
+                    recognitionEnd.set_value(std::optional<std::string>{ error.str() }); // Notify to stop recognition.
+                }
+                else
+                {
+                    std::ostringstream error;
+                    error << "Request was cancelled for an unrecognized reason: " << (int)e.Reason << std::endl;
+                    recognitionEnd.set_value(std::optional<std::string>{ error.str() }); // Notify to stop recognition.
+                }
+            });
+
+        speechRecognizer->SessionStopped.Connect([this, &recognitionEnd](const SessionEventArgs& e)
+            {
+                WriteToConsole("Session stopped.\n");
+                recognitionEnd.set_value(std::nullopt); // Notify to stop recognition.
+            });
+
+        // Starts continuous recognition. Uses StopContinuousRecognitionAsync() to stop recognition.
+        speechRecognizer->StartContinuousRecognitionAsync().get();
+
+        // Waits for recognition end.
+        auto result = recognitionEnd.get_future().get();
+
+        // Stops recognition.
+        speechRecognizer->StopContinuousRecognitionAsync().get();
+        
+        return result;
+    }
+};
+
+int main(int argc, char* argv[])
+{
+    const std::string usage = "Usage: caption.exe [...]\n\n"
+"  CONNECTION\n"
+"    --key KEY                     Your Azure Speech service subscription key.\n"
+"    --region REGION               Your Azure Speech service region.\n"
+"                                  Examples: westus, eastus\n\n"
+"  LANGUAGE\n"
+"    --languages LANG1,LANG2       Enable language identification for specified languages.\n"
+"                                  Example: en-US,ja-JP\n\n"
+"  INPUT\n"
+"    --input FILE                  Input audio from file (default input is the microphone.)\n"
```

Comment:
If if this is supposed to be modeled after the CLI then it should be `--file FILE`

Reply:
True but we also have possible output to a file. I could use --file for input and --output for output but that feels asymmetric.
I would use --output file FILE but the command line arg parsing in the sample only supports one param to each flag. --output_file FILE seems like a pain for the user to type.